### PR TITLE
Csharp extern improvements

### PR DESCRIPTION
**Pull request by RustanLeino**
Improved support for mixing Dafny and C#.

### CODE REVIEW

File: Test/comp/Extern4.go [hunk 1]
```diff
@@ -1,27 +1,53 @@
 package Library
```
Comment (seanmcl) [inline Test/comp/Extern4.go]: Why is this change in is PR?

File: Test/comp/Extern3.js [hunk 2]
```diff
@@ -15,5 +15,17 @@ let Library = (function() {
     }
   };
   
+  $module.Mixed = class Mixed {
```
Comment (seanmcl) [inline Test/comp/Extern3.js]: Does this change belong in this PR?

File: Test/comp/Extern.dfy [hunk 3]
```diff
@@ -16,4 +22,18 @@ module {:extern "Library"} Library {
...
+    static function Seven(): int { 7 }
+    static method M() { print "AllDafny.M\n"; }
+  }
+  class {:extern} Mixed {
+    static function Seven(): int { 7 }
+    static method M() { print "Mixed.M\n"; }
+    static method {:extern} P()
+  }
+  class {:extern} AllExtern {
+    static function Seven(): int { 7 }
```
Comment (seanmcl) [inline Test/comp/Extern.dfy]: Should this be `extern`?  (Given the name `AllExtern`)

File: Test/comp/Extern4.go (hunk 1 above)
Comment (RustanLeino) [reply Test/comp/Extern4.go]: I goofed. I usually run the test suite on my local machine, but thought I'd let Travis take care of it. But then I forgot to check the status before doing the pull request. It should have been part of the original pull request.

File: Test/comp/Extern.dfy (hunk 3 above)
Comment (RustanLeino) [reply Test/comp/Extern.dfy]: It's ghost, so it doesn't get compiled. (I suppose the name of the class could have been `AllExternOrGhost`.)

File: Test/comp/Extern3.js (hunk 2 above)
Comment (RustanLeino) [reply Test/comp/Extern3.js]: Yes, the test file Extern.dfy is run with the C#, JavaScript, and Go compilers, so my change in Extern.dfy required me to also fill in handwritten code in all of Extern2.cs, Extern3.js, and Extern4.go.

**Pull request merged**